common_types/src/schema.rs

```diff
@@ -380,7 +381,7 @@ pub(crate) struct ColumnSchemas {
     columns: Vec<ColumnSchema>,
     /// Column name to index of that column schema in `columns`, the index is
     /// guaranteed to be valid
-    name_to_index: HashMap<String, usize>,
+    name_to_index: ArrayMap<String, usize>,
```

Comment:
If the cost comes from the hash computing, `BTreeMap` may work better?

Reply:
I don't know how node inside BTreeMap is stored, ArrayMap is optimized for `get` when map size is small, array is the best structure I can think of.